clang/include/clang/InstallAPI/Context.h

```diff
@@ -35,19 +36,19 @@ struct InstallAPIContext {
   /// Active target triple to parse.
   llvm::Triple TargetTriple{};
 
-  /// Output stream to write TextAPI file to.
-  std::unique_ptr<llvm::raw_pwrite_stream> OS = nullptr;
-
-  /// DiagnosticsEngine to report errors.
-  llvm::IntrusiveRefCntPtr<DiagnosticsEngine> Diags = nullptr;
-
   /// File Path of output location.
   StringRef OutputLoc{};
 
   /// What encoding to write output as.
   llvm::MachO::FileType FT = llvm::MachO::FileType::TBD_V5;
 };
 
+class InstallAPIAction : public ASTFrontendAction {
```

Comment:
I would move this part into a dedicated header file.

Reply:
Removed this entirely for now. Will add back in follow-up where the header input processing + frontend are hooked together.